openid-4-verifiable-presentations-1_0.md

```diff
@@ -746,7 +752,15 @@ The following is a non-normative example of the payload of the JWT used in the e
 
 <{{examples/response/jarm_jwt_vc_json_body.json}}
 
-## Error Response
+## Transaction Data {#transaction_data}
+
+Transaction data mechanism enables binding between the user identification/authentication and user’s authorization, for example to complete a payment transaction, or to sign specific document(s) using QES (Qualified Electronic Signatures). This is achieved by signing the transaction data used for user authorization with the user-controlled key used for proof of possession of the Credential being presented as a means for user identification/authentication.
+
+The Wallet that received `transaction_data` parameter in the request, MUST include in the response a `transaction_data` parameter defined below: 
+
+* `transaction_data`: Array of hashes, where each hash is calculated using a hash function over the strings received in the `transaction_data` request parameter. Each hash value ensures the integrity of, and maps to, the respective transaction data object. Where in the response this parameter is included is defined by each Credential Format Profile, but it has to be included in the  mechanism used for the proof of possession of the Credential that is signed using the user-controlled key.
```

Comment:
I think I might know where the words "ensures the integrity of, and maps to, the respective" came from :) 

I know one is in the request and one is somewhere in the response but using the same name `transaction_data` for the things in the request and the hashes of those things in the response feels erroneous. Maybe something like `tx_data_hashes` for this response one. Or because it has to be defined by each Credential Format Profile anyway, let the name be defined there too. Maybe.

Reply:
good point! I think for interop purposes, this parameter should be defined in VP

```suggestion
* `tx_data_hashes`: Array of hashes, where each hash is calculated using a hash function over the strings received in the `transaction_data` request parameter. Each hash value ensures the integrity of, and maps to, the respective transaction data object. Where in the response this parameter is included is defined by each Credential Format Profile, but it has to be included in the  mechanism used for the proof of possession of the Credential that is signed using the user-controlled key.
```

Reply:
That suggested change https://github.com/openid/OpenID4VP/pull/197#discussion_r1709406435 works for me. But note it would also need a change to what is currently line on 759.